src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -5892,6 +5892,12 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="WRN_UnconsumedEnumeratorCancellationAttributeUsage_Title" xml:space="preserve">
     <value>The EnumeratorCancellationAttribute will have no effect. The attribute is only effective on a parameter of type CancellationToken in an async-iterator method returning IAsyncEnumerable</value>
   </data>
+  <data name="WRN_UndecoratedCancellationTokenParameter" xml:space="preserve">
+    <value>Async-iterator '{0}' has one or multiple parameters of type 'CancellationToken' but none of them is decorated with the 'EnumeratorCancellation' attribute, so the cancellation token parameter from the generated 'IAsyncEnumerable&lt;>.GetAsyncEnumerator' will be unconsumed</value>
```

Comment:
>so the cancellation token parameter from the generated 'IAsyncEnumerable&lt;>.GetAsyncEnumerator' will be unconsumed [](start = 166, length = 116)

This seems a little confusing since it describes the generated code rather than the source.

Is the issue that the undecorated parameter from source will be ignored? If so, perhaps the warning message should say: "... so the cancellation token parameters will be ignored."

Reply:
Here's what `[EnumeratorCancellation]` means: in the body of the async-iterator, you will see the parameter passed to the method, unless a non-default token is passed via `GetAsyncEnumerator` (or indirectly via `WithCancellation` wrapper) in which case you will get that value instead.